src/Formatters/ImportsFormatter.cs

```diff
@@ -0,0 +1,53 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the MIT license.  See License.txt in the project root for license information.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Formatting;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.CodeAnalysis.Tools.Formatters
+{
+    /// <summary>
+    /// ImportsFormatter that uses the <see cref="Formatter"/> to format document import directives.
+    /// </summary>
+    internal sealed class ImportsFormatter : DocumentFormatter
+    {
+        protected override string FormatWarningDescription => Resources.Fix_imports_ordering;
+        private readonly DocumentFormatter _endOfLineFormatter = new EndOfLineFormatter();
+
+        internal override async Task<SourceText> FormatFileAsync(
+            Document document,
+            SourceText sourceText,
+            OptionSet optionSet,
+            AnalyzerConfigOptions? analyzerConfigOptions,
+            FormatOptions formatOptions,
+            ILogger logger,
+            CancellationToken cancellationToken)
+        {
+            try
+            {
+                var organizedDocument = await Formatter.OrganizeImportsAsync(document, cancellationToken);
+                if (organizedDocument == document)
```

Comment:
I didn't think equality for documents was implemented. Did we add that? I would normally compare the document ids.

Reply:
Well, I just want to know if the document was changed, so I thought a ref equals would be fine. Could use `HasTextChanged` instead tho.

Reply:
yeah, `HasTextChanged` seems the most efficient implementation since it looks at the text version

Reply:
And it is internal. =) I could run GetTextChangesAsync and check to see if it is empty.

Reply:
Actually, I've verified the ref check is doing exactly what I want it to do. The call to Organize is returning the same document instance when there are no changes.

Reply:
ah, that's why I always recommend customers compare `DocumentId`s, I forgot. `DocumentId` implements equitable so you can do `(organizedDocument.Id == document.Id)`

Reply:
> The call to Organize is returning the same document instance when there are no changes.

Good to know. Feel free to ignore this then. My only nit is that I think this behavior is an implementation detail and not part of the public API contract.